attestation/git/git.go

```diff
@@ -113,19 +219,57 @@ func (a *Attestor) Attest(ctx *attestation.AttestationContext) error {
 }
 
 func (a *Attestor) Subjects() map[string]cryptoutil.DigestSet {
+	subjects := make(map[string]cryptoutil.DigestSet)
+
 	subjectName := fmt.Sprintf("commithash:%v", a.CommitHash)
-	return map[string]cryptoutil.DigestSet{
-		subjectName: {
-			{
-				Hash:   crypto.SHA1,
-				GitOID: true,
-			}: a.CommitHash,
-		},
+	subjects[subjectName] = cryptoutil.DigestSet{
+		{
+			Hash:   crypto.SHA1,
+			GitOID: false,
+		}: a.CommitHash,
+	}
+
+	//add author email
+	subjectName = fmt.Sprintf("authoremail:%v", a.AuthorEmail)
+	ds, err := cryptoutil.CalculateDigestSetFromBytes([]byte(a.AuthorEmail), a.hashes)
+	if err != nil {
+		return nil
 	}
+
+	subjects[subjectName] = ds
+
+	//add committer email
+	subjectName = fmt.Sprintf("committeremail:%v", a.CommitterEmail)
+	ds, err = cryptoutil.CalculateDigestSetFromBytes([]byte(a.CommitterEmail), a.hashes)
+	if err != nil {
+		return nil
+	}
+
+	subjects[subjectName] = ds
+
+	//add parent hashes
+	for _, parentHash := range a.ParentHashes {
+		subjectName = fmt.Sprintf("parenthash:%v", parentHash)
```

Comment:
we might not want this as a subject.  Do we want to pull up children when we search for parents?  We can always filter on name, but that adds complications.

Reply:
We should be in the habit of always filtering by name as well as digest when searching archivista for subjects.

The only use case of having the parents as subjects here I can think of is to find children of a commit we're currently looking at, since we already have the parents on the attestor itself.  That could have some value, potentially.